test/extended/machines/cluster.go

```diff
@@ -100,4 +119,293 @@ var _ = g.Describe("[sig-node] Managed cluster", func() {
+				{
+					Name: "list-boots",
+					SecurityContext: &corev1.SecurityContext{
+						RunAsUser:  &zero,
+						Privileged: &isTrue,
+					},
+					Image: image.ShellImage(),
+					Command: []string{
+						"/bin/bash",
+						"-c",
+						command,
+					},
+					TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/host",
+							Name:      "host",
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if apierrors.IsAlreadyExists(err) {
+		return getNumberOfBootsForNode(kubeClient, nodeName, attempt+1)
+	}
+	if err != nil {
+		return 0, fmt.Errorf("failed to create pod %s: %v", name, err)
+	}
+
+	podName := pod.Name
+
+	// Wait for pod to run
+	err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+		podGet, getErr := kubeClient.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+		if getErr != nil {
+			return false, getErr
+		}
+		switch podGet.Status.Phase {
+		case corev1.PodSucceeded:
+			return true, nil
+		case corev1.PodFailed:
+			return true, fmt.Errorf("journalctl command in pod %s failed. Pod status: %#v", name, podGet.Status)
+		default:
+			return false, nil
+		}
+	})
+	if err != nil {
+		return 0, fmt.Errorf("pod %s failed to complete: %v", name, err)
+	}
+
+	// Fetch pod logs
+	linesInPodLogs := -1
+	podLogOpts := corev1.PodLogOptions{}
+	req := kubeClient.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &podLogOpts)
+	podLogs, err := req.Stream(ctx)
+	if err != nil {
+		return linesInPodLogs, fmt.Errorf("failed to open stream to read %s pod logs: %v", name, err)
+	}
+	defer podLogs.Close()
+
+	// Count number of boots
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return linesInPodLogs, fmt.Errorf("failed to copy information from podLogs to buf: %v", err)
+	}
+	linesInPodLogs = strings.Count(buf.String(), "\n") - 1
+	if err != nil {
+		return linesInPodLogs, fmt.Errorf("failed to fetch boot list from %s pod logs: %v", name, err)
+	}
+	return linesInPodLogs, nil
+}
+
+func createTestBed(ctx context.Context, oc *exutil.CLI) {
+	err := callProject(ctx, oc, true)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callServiceAccount(ctx, oc, true)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callRBAC(ctx, oc, true)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = exutil.WaitForServiceAccountWithSecret(
+		oc.AdminKubeClient().CoreV1().ServiceAccounts(namespace),
+		serviceAccountName)
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+func deleteTestBed(ctx context.Context, oc *exutil.CLI) {
+	err := callRBAC(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callServiceAccount(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+	err = callProject(ctx, oc, false)
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+func callRBAC(ctx context.Context, oc *exutil.CLI, create bool) error {
+	obj := &authv1.ClusterRoleBinding{
```

Comment:
If this were a RoleBinding, it looks like all of the test fixtures would be namespace-scoped and you could rely on namespace finalization to clean them up.

Reply:
good idea